lib/mayaUsd/ufe/UsdUndoInsertChildCommand.cpp

```diff
@@ -64,9 +76,21 @@ UsdUndoInsertChildCommand::UsdUndoInsertChildCommand(
     }
     fUsdDstPath = parent->prim().GetPath().AppendChild(TfToken(childName));
 
-    fLayer = MayaUsdUtils::defPrimSpecLayer(childPrim);
-    if (!fLayer) {
-        std::string err = TfStringPrintf("No prim found at %s", childPrim.GetPath().GetString().c_str());
+    fChildLayer = MayaUsdUtils::defPrimSpecLayer(childPrim);
+    if (!fChildLayer) {
+        std::string err = TfStringPrintf("No child prim found at %s", childPrim.GetPath().GetString().c_str());
+        throw std::runtime_error(err.c_str());
+    }
+
+    auto parentPrim = parent->prim();
+    // If parent prim is the pseudo-root, no def primSpec will be found, so
+    // just use the edit target layer.
+    fParentLayer = parentPrim.IsPseudoRoot() ?
```

Comment:
I guess this is expected, but I didn't know about it until I ran into the issue during testing.